packages/flutter/lib/src/material/text_field.dart

```diff
@@ -68,8 +68,21 @@ class TextField extends StatefulWidget {
   /// The [maxLines] property can be set to null to remove the restriction on
   /// the number of lines. By default, it is one, meaning this is a single-line
   /// text field. [maxLines] must not be zero. If [maxLines] is not one, then
-  /// [keyboardType] is ignored, and the [TextInputType.multiline] keyboard type
-  /// is used.
+  /// [keyboardType] is ignored, and the [TextInputType.multiline] keyboard
+  /// type is used.
+  ///
+  /// The [maxLength] property is set to null by default, which means the
+  /// number of characters allowed in the text field is not restricted. If
+  /// [maxLength] is set, it will display a character counter below the field,
+  /// showing how many characters have been entered, how many are allowed,
+  /// and will prevent any characters above the limit from being entered. It
```

Comment:
This could be a separate sentence: that maybe clarifies what happens upon overflow: After [maxLength] characters have been input, additional input is ignored (I assume we're dropping the overflow characters on the floor).

Also: "It" could be the TextField